Svc/DpCatalog/DpCatalog.cpp

```diff
@@ -534,6 +455,134 @@ Fw::CmdResponse DpCatalog::fillBinaryTree() {
 
 }  // end fillBinaryTree()
 
+int DpCatalog::processFile(Fw::String fullFile, FwSizeType dir = DP_MAX_DIRECTORIES) {
+    // file class instance for processing files
+    Os::File dpFile;
+
+    // Working buffer for DP headers
+    U8 dpBuff[Fw::DpContainer::MIN_PACKET_SIZE];  // Header buffer
+    Fw::Buffer hdrBuff(dpBuff, sizeof(dpBuff));   // buffer for container header decoding
+    Fw::DpContainer container;                    // container object for extracting header fields
+
+    this->log_ACTIVITY_LO_ProcessingFile(fullFile);
+
+    // Check if file is in one of our directories
+    if (dir >= DP_MAX_DIRECTORIES) {
+        // Grab the directory string (up until the final slash)
+        // Could be found directly w/ a dirname func or regex
+        FwSignedSizeType loc = Fw::StringUtils::substring_find_last(
+            fullFile.toChar(), fullFile.length(), DIRECTORY_DELIMITER,
+            Fw::StringUtils::string_length(DIRECTORY_DELIMITER, sizeof(DIRECTORY_DELIMITER)));
+
+        // TODO: What happens w/ relative paths and root dir files
+        if (-1 == loc) {
+            this->log_WARNING_HI_DirectoryOpenError(fullFile, Os::FileSystem::Status::OTHER_ERROR);
+            return 0;
+        }
+
+        for (dir = 0; dir < this->m_numDirectories; dir++) {
+            const char* const dir_string = this->m_directories[dir].toChar();
+
+            // Compare both strings up to location of final slash
+            // memsafe since both are fixed width strings
+            // and loc is before the fixed width
+            if (strncmp(dir_string, fullFile.toChar(), static_cast<FwSizeType>(loc)) == 0) {
+                break;
+            }
+        }
+
+        if (dir == this->m_numDirectories) {
+            this->log_WARNING_HI_DirectoryOpenError(fullFile, Os::FileSystem::Status::OTHER_ERROR);
```

Comment:
Create a new event specifically for this, perhaps something like "Unable to add file {}; directory not managed"